README.md

```diff
@@ -1,5 +1,97 @@
 # Fixed Size Numbers
 
-[WIP] Implementing size enforcement of BigNumbers in TypeScript.
+A functional implementation of fixed size integers in javascript with typing in TypeScript.
 
-The goal of this library is to provide simple safe math checks to enforce sizes on signed and unsigned integers.  
\ No newline at end of file
+>Note: this library is just a typed wrapper on [BigNumber.js](https://github.com/MikeMcl/bignumber.js/)
+
+**Currently no Floating Points or Decimal values are supported!!!**
+
+## API
+
+Create a new signed or unsigned integer by passing a value in string, number, or BigNumber format to the constructor function
+
+Access the value and the size with their respective attributes (_value, and _size)
+
+>Note: _value is always a BigNumber
+
+Or preform safe math using the methods provided
+
+```javascript
+import { Uint8, Int8 } from 'fixed-sized-numbers-ts'
+let safe72 = Uint8(72)
+let safeNegative54 = Int(-54)
+
+console.log(safe72)
+/*
+{
+    _value: 72,
+    _size: 8,
+    validateSize: [Function],
+    add: [Function],
+    sub: [Function],
+    mul: [Function],
+    div: [Function],
+    _uint8: true
+}
+*/
+
+console.log(safeNegative54)
+/*
+{
+    _value: -54,
+    _size: 8,
+    validateSize: [Function],
+    add: [Function],
+    sub: [Function],
+    mul: [Function],
+    div: [Function],
+    _int8: true,
+    _isPositive: false
+}
+*/
+```
+
+>Note: Uint64 and Int64 need to be constructed using strings because of imprecission in Javascript after 53 bits, this is enforced in the typing
+
+```javascript
+import { Uint64 } from 'fixed-sized-numbers-ts'
+let bigSafeNumber = Uint64("18446744073709551616")
+
+console.log(bigSafeNumber)
+/*
+{
+    _value: "18446744073709551616",
+    _size: 8,
+    validateSize: [Function],
+    add: [Function],
+    sub: [Function],
+    mul: [Function],
+    div: [Function],
+    _uint8: true
+}
+*/
+```
+
+For safe math the first value in the equation will be the one who's method is being called, the second number will be the number passed to it.
+
+>Note: Types must match; if you are calling the method from a Uint8 on a Uint8 it will work, but if you are calling it on a Uint16 it will not, this is enforced at the type level. Another way to think about this is that the math methods return a value with the same type that it belongs to i.e. all Uint8 methods return Uint8 objects.
```

Comment:
Consider revising this to be more to the point. You might also want to note uint8 + int8